pkg/solana/logpoller/log_poller.go

```diff
@@ -2,40 +2,62 @@ package logpoller
 
 import (
 	"context"
+	"encoding/base64"
 	"errors"
+	"fmt"
+	"math"
 	"time"
 
+	"github.com/lib/pq"
 	"github.com/smartcontractkit/chainlink-common/pkg/logger"
 	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink-common/pkg/utils"
+
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/client"
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/internal"
 )
 
 var (
 	ErrFilterNameConflict = errors.New("filter with such name already exists")
 )
 
 type ORM interface {
+	ChainID() string
 	InsertFilter(ctx context.Context, filter Filter) (id int64, err error)
 	SelectFilters(ctx context.Context) ([]Filter, error)
 	DeleteFilters(ctx context.Context, filters map[int64]Filter) error
 	MarkFilterDeleted(ctx context.Context, id int64) (err error)
 	MarkFilterBackfilled(ctx context.Context, id int64) (err error)
+	InsertLogs(context.Context, []Log) (err error)
+	SelectSeqNums(ctx context.Context) (map[int64]int64, error)
+}
+
+type ILogPoller interface {
+	Start(context.Context) error
+	Close() error
+	RegisterFilter(ctx context.Context, filter Filter) error
+	UnregisterFilter(ctx context.Context, name string) error
+	Process(programEvent ProgramEvent) error
 }
 
 type LogPoller struct {
```

Comment:
This really should be something like `Service` so that imports into other packages read as `logpoller.Service`. Using `logpoller.LogPoller` is repetative. I understand it is in an effort to be more like the `EVM` implementation, but we shouldn't copy something that is wrong for the sake of being consistently wrong.